>From 526864fc49542c8be0a117ae39c45902f86a4ccd Mon Sep 17 00:00:00 2001
From: Richard Ash <richard@audacityteam.org>
Date: Fri, 16 Aug 2013 22:40:15 +0100
Subject: [PATCH] Add support for Album and Track tags in RIFF files
 Add support for Album and Track tags in RIFF files in the LIST chunk of WAV files
 as per other formats. Patch by Joel Bouchat <bouchat@hotmail.com> against
 libsndfile 1.0.24, forward-ported by Richard Ash

---
 src/wav.c | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/src/wav.c b/src/wav.c
index daaba39..d28b18d 100644
--- a/src/wav.c
+++ b/src/wav.c
@@ -104,6 +104,7 @@
 #define ISBJ_MARKER		(MAKE_MARKER ('I', 'S', 'B', 'J'))
 #define ICMT_MARKER		(MAKE_MARKER ('I', 'C', 'M', 'T'))
 #define IAUT_MARKER		(MAKE_MARKER ('I', 'A', 'U', 'T'))
+#define ITRK_MARKER		(MAKE_MARKER ('I', 'T', 'R', 'K'))
 
 /* Weird WAVPACK marker which can show up at the start of the DATA section. */
 #define wvpk_MARKER (MAKE_MARKER ('w', 'v', 'p', 'k'))
@@ -1253,6 +1254,14 @@ wav_write_strings (SF_PRIVATE *psf, int location)
 				psf_binheader_writef (psf, "ms", IGNR_MARKER, psf->strings.storage + psf->strings.data [k].offset) ;
 				break ;
 
+			case SF_STR_ALBUM :
+				psf_binheader_writef (psf, "ms", IPRD_MARKER, psf->strings.storage + psf->strings.data [k].offset) ;
+				break ;
+
+			case SF_STR_TRACKNUMBER :
+				psf_binheader_writef (psf, "ms", ITRK_MARKER, psf->strings.storage + psf->strings.data [k].offset) ;
+				break ;
+
 			default :
 				break ;
 			} ;
@@ -1384,6 +1393,7 @@ wav_subchunk_parse (SF_PRIVATE *psf, int chunk, uint32_t length)
 			case ISBJ_MARKER :
 			case ISRC_MARKER :
 			case IAUT_MARKER :
+			case ITRK_MARKER :
 					bytesread += psf_binheader_readf (psf, "4", &dword) ;
 					dword += (dword & 1) ;
 					if (dword >= SIGNED_SIZEOF (buffer))
@@ -1477,6 +1487,12 @@ wav_subchunk_parse (SF_PRIVATE *psf, int chunk, uint32_t length)
 			case IGNR_MARKER :
 					psf_store_string (psf, SF_STR_GENRE, buffer) ;
 					break ;
+			case IPRD_MARKER :
+					psf_store_string (psf, SF_STR_ALBUM, buffer) ;
+					break ;
+			case ITRK_MARKER :
+					psf_store_string (psf, SF_STR_TRACKNUMBER, buffer) ;
+					break ;
 			} ;
 		} ;
 
-- 
1.8.1.5

